test/unit/deployment/test_language_container_deployer.py

```diff
@@ -34,49 +34,51 @@ def mock_pyexasol_conn() -> ExaConnection:
-                                         bucketfs_location=mock_bfs_location)
+                                         bucketfs_path=create_autospec(bfs.path.PathLike))
 
     deployer.upload_container = MagicMock()
     deployer.activate_container = MagicMock()
     return deployer
 
 
 def test_slc_deployer_deploy(container_deployer, container_file_name, container_file_path):
-    container_deployer.run(container_file=container_file_path, bucket_file_path=container_file_name, alter_system=True,
+    container_deployer.run(container_file=container_file_path,
+                           bucket_file_path=container_file_name,
+                           alter_system=True,
                            allow_override=True)
-    container_deployer.upload_container.assert_called_once_with(container_file_path, container_file_name)
-    container_deployer.activate_container.assert_called_once_with(container_file_name, LanguageActivationLevel.System,
+    container_deployer.upload_container.assert_called_once_with(container_file_path,
+                                                                container_file_name)
+    container_deployer.activate_container.assert_called_once_with(container_file_name,
+                                                                  LanguageActivationLevel.System,
                                                                   True)
 
 
 def test_slc_deployer_upload(container_deployer, container_file_name, container_file_path):
     container_deployer.run(container_file=container_file_path, alter_system=False)
-    container_deployer.upload_container.assert_called_once_with(container_file_path, container_file_name)
+    container_deployer.upload_container.assert_called_once_with(container_file_path,
+                                                                container_file_name)
     container_deployer.activate_container.assert_not_called()
 
 
 def test_slc_deployer_activate(container_deployer, container_file_name):
     container_deployer.run(bucket_file_path=container_file_name, alter_system=True, allow_override=True)
     container_deployer.upload_container.assert_not_called()
-    container_deployer.activate_container.assert_called_once_with(container_file_name, LanguageActivationLevel.System,
+    container_deployer.activate_container.assert_called_once_with(container_file_name,
+                                                                  LanguageActivationLevel.System,
                                                                   True)
 
 
+@patch('exasol.python_extension_common.deployment.language_container_deployer.get_udf_path')
 @patch('exasol.python_extension_common.deployment.language_container_deployer.get_language_settings')
-def test_slc_deployer_generate_activation_command(mock_lang_settings, container_deployer, language_alias,
+def test_slc_deployer_generate_activation_command(mock_lang_settings, mock_udf_path,
```

Comment:
What is the order how the patch mocks get passed to the test?

Because, naively, I would assume, mock_lang_settings is the first patch decorator (get_udf_path) and mock_udf_path the second (get_language_settings).

Reply:
No, it's the opposite order, a bit counter-intuitive.

Reply:
ah ok, I had almost guessed that way, because decorator wrapping functions

`@patch('exasol.python_extension_common.deployment.language_container_deployer.get_language_settings` probably wraps `test_slc_deployer_generate_activation_command` and 
`@patch('exasol.python_extension_common.deployment.language_container_deployer.get_udf_path')` wraps the results from the first result